src/test/java/com/u1/user/integrationtest/UserIntegrationTest.java

```diff
@@ -113,4 +115,60 @@ public class UserIntegrationTest {
 
     }
 
+    /*-------------------------------登録処理-------------------------------------------------------------------------*/
+    @Test
+    @DataSet(value = "datasets/users.yml")
+    @ExpectedDataSet(value = "datasets/insertUsers.yml", ignoreCols = "id")
+    @Transactional
+    void 名前と生年月日を未使用のIDで紐づけし登録すること() throws Exception {
+        String requestBody = """
+                {
+                           "name": "Tom",
+                           "birthday": "1990/01/01"
+                }
+                """;
+        mockMvc.perform(MockMvcRequestBuilders.post("/users")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(requestBody))
+                .andExpect(MockMvcResultMatchers.status().isCreated())
+                .andExpect(MockMvcResultMatchers.content().json("""
+                        {
+                            "message": "user created"
+                        }
+                        """));
+    }
+
+    @Test
+    @DataSet(value = "datasets/users.yml")
+    @Transactional
+    void 指定しないフォーマットで登録しようとしたとき例外処理を返すこと() throws Exception {
```

Comment:
指定しないフォーマットというよりは不正なフォーマットでしょうか。
それから、HTTPレスポンスを返すわけですから例外という言葉を使うのは不適切です。
```suggestion
    void 不正なフォーマットで登録しようとしたとき400エラーを返すこと() throws Exception {
```

Reply:
テスト名変更しました！

https://github.com/u-1pena/assignment10/commit/87ce9601040b11bcdde34f8f77ca901036e46ecb
https://github.com/u-1pena/assignment10/blob/87ce9601040b11bcdde34f8f77ca901036e46ecb/src/test/java/com/u1/user/integrationtest/UserIntegrationTest.java#L150-L154

Reply:
[24cd2e7](https://github.com/u-1pena/assignment10/pull/14/commits/24cd2e7d9af309248ddef67701d2d6097e06be0c)

例外処理を誤解していました。
ExceptionHandlerを返すから例外処理と思い込んでいました。

Chat GPTより
例外処理とは、プログラムが実行中に何らかの異常状況が発生した際に、その異常状況に対処するための仕組みです。例外処理は、プログラムの流れを中断させずにエラーを処理し、適切な処理を行うための仕組みです。一方で、HTTPレスポンスは、クライアントに対してリクエストの結果を通知するためのものであり、エラーの発生を通知する際に使用されることもありますが、HTTPレスポンスそのものが例外処理というわけではありません。

単体テストの部分は例外を返すまでの処理なので例外処理であり、結合テストは例外処理を経ているが、ここでのテストはHTTPレスポンスのエラーを知らせているので例外処理ではない。という認識でよろしいでしょうか？